### ISSUE TITLE
Onchain funds not showing after restore

### ISSUE THREAD

**Issue by Sjors** [LN-Zap/zap-iOS#230]
**Describe the bug**
I was smart enough to delete the mainnet app instead of the testnet app. Fortunately I still had the recovery phrase. I re-entered it (nice UX!), but then no coins showed up. Part of the funds were in a channel (I'll figure that out later), but part of the funds were on chain.

**To Reproduce**
I haven't tried reproducing, but:
1. Install from Testflight
2. Fund, wait for confirmation
3. Open a channel with some of the funds
4. Leave the app, have fun on Twitter
5. (not sure about this) don't open the app when funds are confirmed 
6. delete the app
7. reinstall the app
8. enter recovery phrase

**Expected behavior**
I would expect my historical on chain transactions to show up and the funds that never entered a channel.

**Screenshots**
Total balance $0, no transaction history.

**Your environment**
- Device: iPhone X
- OS: iOs 12.4
- Version 0.4.0 (23) mainnet

**Additional context**
The app shows a channel.backup file from 11:24 that I made a copy of. In my iCloud drive I also found a channel.backup file from 11:21 (which I also copied). The new file is 45 bytes, the old file 423 bytes. So my guess is I might even be able to [recover](https://medium.com/@rahil471/enable-channel-backups-and-fund-recovery-on-lnd-lightning-network-3f27be42eb43) the channel (I also control the node on the other side). It's in a folder `Zap/0....2` which is from August 1. So it might also be the testnet backup. I don't see a seperate `Zap Mainnet` folder in iCloud.

**Comment by Sjors** [comment]
I deleted the app again and this time manually selected the 423 byte backup file during the recovery process. After loading transactions it briefly showed my balance as $0 and then crashed (I assume you got the crash logs). Starting the app again, still no funds.

I did some digging in the lnd log. It says `Found birthday block: height=591126` which is before my funding transaction. But then it says `Started rescan from block 000...ce8b (height 591126) for 0 addresses`, which suggests a problem.

Then I think it tries and fails to restore the backup (which might be the wrong one):
```
2019-08-24 09:59:29.247 [ERR] LTND: Unable to start server: unable to unpack chan backup: chacha20poly1305: message authentication failed
```

**Comment by JimmyMow** [comment]
Hey @Sjors.

Thanks for the report. Looking into this currently, hoping to deploy a fix soon

**Comment by Sjors** [comment]
Is this in the new 0.4.0 (24) testflight release? It's probably useful to tag releases on Github, so even without release notes people can see which commits are new.

**Comment by ottosuess** [comment]
sorry for the slow fix and bad communication on this one.
it is in 0.4.0 (24) and you need to do a fresh recovery.

**Comment by Sjors** [comment]
@ottosuess thanks. It found my on chain funds!

Unforunately I couldn't use the backup file (and the crashed when it tried), so maybe that was indeed an older testnet backup.

> 2019-09-02 19:46:29.130 [ERR] LTND: Unable to start server: unable to unpack chan backup: chacha20poly1305: message authentication failed

I only see one Zap/02..... folder on my phone when I use the Files app. Does every new install erase the previous one?

Do you know if it's safe to force close the channel from my c-lightning node? Or would Lnd be unable to sweep its side without a correct backup file?

I would say the 2500 lookahead window is overkill when combined with BIP158 filters. It causes a lot of false positive block downloads. It seems unlikely that a mobile app users generates more than e.g. 100 addresses without using them. In fact, the current UI won't even let you generate more than 1.

**Comment by Sjors** [comment]
I also tried recovering with the 45 byte backup I mentioned earlier. This did not crash, but also did not find my channel.

When you delete the app from the phone, the local backup is also deleted; the Files app will no longer show a Zap directory on the local device.

I also don't see the backups on iCloud drive; there is a Zap folder there with a backup directory but hasn't been updated today even though I reinstalled the app a few times. I do see a green circle next to iCloud in the channel backup screen. Is it somewhere else in the filesystem? I'll make a separate ticket for that.
